src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs

```diff
@@ -70,15 +70,7 @@ public async Task StartAsync(CancellationToken cancellationToken = default)
 
             if (_options.ServicesStartConcurrently)
             {
-                Task tasks = Task.WhenAll(_hostedServices.Select(async service =>
-                {
-                    await service.StartAsync(combinedCancellationToken).ConfigureAwait(false);
-
-                    if (service is BackgroundService backgroundService)
-                    {
-                        _ = TryExecuteBackgroundServiceAsync(backgroundService);
-                    }
-                }));
+                Task tasks = Task.WhenAll(_hostedServices.Select(service => Task.Run(() => StartAndTryToExecuteAsync(service, combinedCancellationToken))));
```

Comment:
Should we be passing `token` into `Task.Run` as well?

```suggestion
                Task tasks = Task.WhenAll(_hostedServices.Select(service => Task.Run(() => StartAndTryToExecuteAsync(service, combinedCancellationToken), combinedCancellationToken)));
```